crates/volta-core/src/tool/yarn/resolve.rs

```diff
@@ -84,8 +86,25 @@ fn fetch_yarn_index() -> Fallible<(String, PackageIndex)> {
 }
 
 fn resolve_custom_tag(tag: String) -> Fallible<Version> {
-    let (url, mut index) = fetch_yarn_index()?;
+    if env::var_os("VOLTA_FEATURE_YARN_3").is_some() {
+        // first try yarn2+, which uses "@yarnpkg/cli-dist" instead of "yarn"
+        let (url, mut index) = fetch_yarn_index("@yarnpkg/cli-dist")?;
+
+        let matches_yarn_2 = VersionReq::parse("2.*").unwrap();
+        if let Some(version) = index.tags.remove(&tag) {
+            debug!("Found yarn@{} matching tag '{}' from {}", version, tag, url);
+            if matches_yarn_2.matches(&version) {
```

Comment:
Similarly here, we can probably check `version.major == 2` directly, rather than needing to create a `VersionReq` to compare.

Reply:
Fixed in db9e6d1d0e1a2ca910231af66ce65b8f9416f382